ui/src/overlays/components/OverlayController.tsx

```diff
@@ -0,0 +1,31 @@
+// Libraries
+import React, {FunctionComponent} from 'react'
+import {connect} from 'react-redux'
+
+// Types
+import {AppState} from 'src/types'
+
+// Components
+import {Overlay} from '@influxdata/clockface'
+import NoteEditorOverlay from 'src/dashboards/components/NoteEditorOverlay'
+
+interface StateProps {
+  overlayID: string
+}
+
+const OverlayController: FunctionComponent<StateProps> = ({overlayID}) => {
+  let activeOverlay = <></>
+
+  switch (overlayID) {
+    case 'add-note':
+    case 'edit-note':
+      activeOverlay = <NoteEditorOverlay />
+      break
+  }
+
+  return <Overlay visible={!!overlayID}>{activeOverlay}</Overlay>
```

Comment:
what do you think about having a boolean representing `visibility` that is explicitly set, rather than relying on a string cast to a boolean? it's more explicit this way - only when the overlayID is `add-note` or `edit-note` and the `activeOverlay` is an actual component will the overlay be visible. currently, if the overlayID is `delete-note` for example, `visible={!!overlayID}` will evaluate to true, even though the `activeOverlay` is a react fragment, not the note editor overlay.

```ts
  let visible = false
  switch (overlayID) {
    case 'add-note':
    case 'edit-note':
      activeOverlay = <NoteEditorOverlay />
      visible = true
      break
  }

  return <Overlay visible={visible}>{activeOverlay}</Overlay>
```

Reply:
ahh that is much cleaner